WordPress/src/main/java/org/wordpress/android/ui/reader/discover/interests/ReaderInterestsViewModel.kt

```diff
@@ -89,49 +102,73 @@ class ReaderInterestsViewModel @Inject constructor(
     }
 
     sealed class UiState(
-        val interestsUiState: List<InterestUiState>,
-        val interests: ReaderTagList,
-        val doneButtonUiState: DoneButtonUiState,
+        open val doneButtonUiState: DoneButtonUiState = DoneButtonHiddenUiState,
         val progressBarVisible: Boolean = false,
         val titleVisible: Boolean = false,
-        val subtitleVisible: Boolean = false
+        val subtitleVisible: Boolean = false,
+        val fullscreenErrorLayoutVisible: Boolean = false
     ) {
-        object InitialUiState : UiState(
-            interestsUiState = emptyList(),
-            interests = ReaderTagList(),
-            doneButtonUiState = DoneButtonHiddenUiState,
+        object LoadingUiState : UiState(
             progressBarVisible = true
         )
 
         data class ContentLoadSuccessUiState(
-            val interestTagsUiState: List<InterestUiState>,
-            val interestTags: ReaderTagList,
-            val doneBtnUiState: DoneButtonUiState
+            val interestsUiState: List<InterestUiState>,
+            val interests: ReaderTagList,
+            override val doneButtonUiState: DoneButtonUiState = DoneButtonDisabledUiState
         ) : UiState(
-            interestsUiState = interestTagsUiState,
-            interests = interestTags,
-            doneButtonUiState = doneBtnUiState,
             progressBarVisible = false,
             titleVisible = true,
-            subtitleVisible = true
+            subtitleVisible = true,
+            fullscreenErrorLayoutVisible = false
         )
 
-        private val checkedInterestsUiState = interestsUiState.filter { it.isChecked }
+        sealed class ContentLoadFailedUiState constructor(
```

Comment:
I'd consider renaming this to something like "FullScreenErrorUiState". The fact that the content load fails doesn't necessarily mean we'll show a progress so the name might be a bit misleading in some scenarios. Wdyt?

Reply:
Renamed here 1097a5a to keep them shorter and consistent with names used in Discover page.